src/auth/AuthRegistry.h

```diff
@@ -0,0 +1,54 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <map>
+#include <vector>
+
+#include "AuthAuthorizeHandler.h"
+#include "AuthMethodList.h"
+#include "common/ceph_mutex.h"
+#include "common/ceph_context.h"
+#include "common/config_cacher.h"
+
+class AuthRegistry : public md_config_obs_t {
+  CephContext *cct;
+  ceph::mutex lock = ceph::make_mutex("AuthRegistry::lock");
```

Comment:
This lock is not reliably used on most of the reader functions? Not sure if they just got missed or you think here’s a writer conflict but the readers are safe for some reason I don’t understand.

I see it’s taken on config refresh and get_handler, but not any of the supported method/mode queries.

Reply:
thanks, fixing!